### PR TITLE
Set Istio latest to Istio 1.7.1.

### LINKED ISSUE #279: Test Istio release "Istio 1.7.1"
Istio recently (2020-09-10 21:26:38 +0000 UTC) released ["Istio 1.7.1"](https://github.com/istio/istio/releases/tag/1.7.1). Let's test it :rocket:.

### PR DESCRIPTION
**Pull request by JRBANCEL**
Fixes #279.
/assign @nak3 @ZhiminXiang

### CODE REVIEW

**Comment by tcnghia** [comment]
I am really excited that we got here.

thanks a lot everyone. especially @nak3 and @howardjohn

File: third_party/istio-1.5.7/install-istio.sh [hunk 1]
```diff
@@ -15,8 +15,8 @@
 # limitations under the License.
 
 # Download and unpack Istio
-ISTIO_VERSION=1.5.7
-ISTIO_TARBALL=istio-${ISTIO_VERSION}-linux.tar.gz
+ISTIO_VERSION=1.7.1
```
Comment (ZhiminXiang) [inline third_party/istio-1.5.7/install-istio.sh:18]: Should this file be under third_party/istio-1.7.1/ ?

**Comment by nak3** [comment]
```
Error: failed to install manifests: generated config failed semantic validation: port http2/80 in gateway cluster-local-gateway invalid: targetPort is set to 0, which requires root. Set targetPort to be greater than 1024 or configure values.gateways.istio-ingressgateway.runAsRoot=true, port https/443 in gateway cluster-local-gateway invalid: targetPort is set to 0, which requires root. Set targetPort to be greater than 1024 or configure values.gateways.istio-ingressgateway.runAsRoot=true
```

@JRBANCEL  We need to change Port numbers for Istio 1.7 like https://github.com/knative/docs/pull/2775/files

**Comment by JRBANCEL** [comment]
@nak3 can you confirm the overlays are not needed anymore.
I looked a the related issues and they seem fixed and `istioctl` fails with the overlays present:

```
overlay for Gateway:istio-ingressgateway does not match any object in output manifest.
overlay for Gateway:cluster-local-gateway does not match any object in output manifest.
```

**Comment by JRBANCEL** [comment]
The installation failure were fixed but now conformance tests are failing. I am looking.

**Comment by nak3** [comment]
> @nak3 can you confirm the overlays are not needed anymore.
> I looked a the related issues and they seem fixed and istioctl fails with the overlays present:

Yes, it is not needed. We can drop it since 1.6 or later.

File: third_party/istio-latest/istio-ci-mesh.yaml [hunk 2]
```diff
@@ -91,17 +81,7 @@ spec:
             ports:
             - port: 15020
               name: status-port
-            - port: 80
+            - port: 8080
```
Comment (nak3) [inline third_party/istio-latest/istio-ci-mesh.yaml:84]: It is still not correct. We need to change port like this:

```
            - port: 80
              targetPort: 8080
              name: http2
            - port: 443
              targetPort: 8443
              name: https
```

**Comment by JRBANCEL** [comment]
@tcnghia @nak3 @ZhiminXiang 
This is finally ready.

**Review by tcnghia — APPROVED** [review]
/lgtm

**Pull request merged**